android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiRecyclerViewAdapter.java

```diff
@@ -7,14 +7,95 @@
 package ti.modules.titanium.ui.widget.listview;
 
 import android.content.Context;
+import android.os.Handler;
+import android.view.LayoutInflater;
 
+import androidx.annotation.NonNull;
 import androidx.recyclerview.selection.SelectionTracker;
+import androidx.recyclerview.widget.DiffUtil;
 import androidx.recyclerview.widget.RecyclerView;
 
-public abstract class TiRecyclerViewAdapter<VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH>
+import org.appcelerator.titanium.proxy.TiViewProxy;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.List;
+
+public abstract class TiRecyclerViewAdapter<VH extends TiRecyclerViewHolder<V>, V extends TiViewProxy>
+	extends RecyclerView.Adapter<VH>
 {
 	protected Context context;
-	private SelectionTracker tracker;
+	protected int id_holder;
+	protected LayoutInflater inflater;
+	protected List<V> models;
+	protected SelectionTracker tracker;
+
+	private Deque<List<V>> pendingUpdates = new ArrayDeque<>();
+
+	public TiRecyclerViewAdapter(@NonNull Context context, @NonNull List<V> models)
+	{
+		// Obtain context.
+		this.context = context;
+
+		// Obtain layout inflater instance.
+		this.inflater = LayoutInflater.from(context);
+
+		// Set models.
+		this.models = models;
+
+		setHasStableIds(true);
+	}
+
+	/**
+	 * Get number of items in list.
+	 *
+	 * @return Integer of item count.
+	 */
+	@Override
+	public int getItemCount()
+	{
+		return this.models.size();
+	}
+
+	/**
+	 * Get unique item identifier.
+	 *
+	 * @param position Index position of item to obtain identifier.
+	 * @return Long of item identifier.
+	 */
+	@Override
+	public long getItemId(int position)
+	{
+		return this.models.get(position).hashCode();
+	}
+
+	/**
+	 * Get list of models.
+	 *
+	 * @return List of models.
+	 */
+	public List<V> getModels()
+	{
+		return this.models;
+	}
```

Comment:
Maybe we shouldn't expose the `models` member variable like this since a caller can add/remove items in the collection, which can cause issues with the `DiffUtil` which is running on a separate thread.

It looks like this method is only called by `ListViewProxy.getCurrentItems()` and nobody calls that method. So we can just get rid of this? Alternatively, we can return the collection as `Iterable<V>` instead to discourage this since it does not provide add/remove methods.